packages/tailwind-typography/src/index.ts

```diff
@@ -0,0 +1,205 @@
+import plugin from 'tailwindcss/plugin'
+
+function isPlainObject(x: unknown): x is Record<string, unknown> {
+  return (
+    typeof x === 'object' &&
+    x?.constructor === Object &&
+    Object.getPrototypeOf(x) === Object.prototype
+  )
+}
+
+export const buildTypographyVariant = (
+  fontFamily: string,
+  fontWeight: number,
+  fontSize: number,
+  lineHeight: number,
+  letterSpacing: number,
+  other?: Record<string, unknown>,
+) => ({
+  fontFamily,
+  fontWeight,
+  fontSize,
+  // Unit-less following http://meyerweb.com/eric/thoughts/2006/02/08/unitless-line-heights/
```

Comment:
**todo:** move this `line-height` comment to the type declaration as JsDoc

So consumers can get more context.

Reply:
Good point! Although I'll be simplifying this even further and remove the `buildTypographyVariant' as it's really not needed.